new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java

```diff
@@ -1491,92 +1528,32 @@ public static V1ServiceList listServices(String namespace) {
-        Boolean.FALSE // Watch for changes to the described resources.
-    );
-    for (V1Service service : v1ServiceList.getItems()) {
-      if (service.getMetadata().getName().equals(serviceName.trim())
-          && service.getMetadata().getNamespace().equals(namespace.trim())) {
-        logger.info("Service Name : " + service.getMetadata().getName());
-        logger.info("Service Namespace : " + service.getMetadata().getNamespace());
-        Map<String, String> labels = service.getMetadata().getLabels();
-        if (labels != null) {
-          for (Map.Entry<String, String> entry : labels.entrySet()) {
-            logger.log(Level.INFO, "Label Key: {0} Label Value: {1}",
-                new Object[]{entry.getKey(), entry.getValue()});
-          }
-        }
-        return service;
-      }
-    }
-    return null;
-  }
-
-  /**
-   * Returns NodePort of a admin server service.
-   *
-   * @param serviceName name of admin server service
-   * @param label key value pair with which the service is decorated with
-   * @param namespace namespace in which to check for the service
-   * @return AdminNodePort of the Kubernetes service if exists else -1
-   * @throws ApiException when there is error in querying the cluster
+   * @param jobBody V1Job object containing job configuration data
+   * @return String job name if job creation is successful
+   * @throws ApiException when create job fails
    */
-  public static int getAdminServiceNodePort(
-      String serviceName,
-      Map<String, String> label,
-      String namespace) throws ApiException {
-
-    V1Service service = getService(serviceName, label, namespace);
-    if (service == null) {
-      logger.info("Could not find the service ${0} in namespace ${1}", serviceName, namespace);
-      return -1;
-    }
-    V1ServiceSpec v1ServiceSpec = service.getSpec();
-    List<V1ServicePort> portList = v1ServiceSpec.getPorts();
-    if (portList == null) {
-      logger.info("Got NULL portList for service ${0} in namespace ${1}", serviceName, namespace);
-      return -1;
-    }
-
-    for (int i = 0; i < portList.size(); i++) {
-      if (portList.get(i).getName().equals("default")) {
-        logger.info(portList.get(i).toString());
-        return portList.get(i).getNodePort().intValue();
-      }
+  public static String createNamespacedJob(V1Job jobBody) throws ApiException {
+    String name = null;
+    String namespace = jobBody.getMetadata().getNamespace();
+    try {
+      BatchV1Api apiInstance = new BatchV1Api(apiClient);
+      V1Job createdJob = apiInstance.createNamespacedJob(
+          namespace, // String | namespace in which to create job
+          jobBody, // V1Job | body of the V1Job containing job data
+          PRETTY, // String | pretty print output.
+          null, // String | dry run or permanent change
+          null // String | field manager who is making the change
+      );
+      name = createdJob.getMetadata().getName();
```

Comment:
null pointer check for createdJob.getMetadata()

Reply:
fixed